smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSink.java

```diff
@@ -168,6 +180,48 @@ private void handleWriteResult(AsyncResult<?> ar, Message<?> message, ProducerRe
                 kafkaHeaders);
     }
 
+    private void createOutgoingTrace(Message<?> message, String topic, int partition, Headers headers) {
+        if (configuration.getTracingEnabled()) {
```

Comment:
This whole block should be externalized in a carefully tested method.

Reply:
Not sure what you mean. It's all in `createOutgoingTrace` method, are you suggesting something else?

Or do you mean there should be unit tests verifying the behavior of just this method?

Reply:
Damned missed it.